mlir/test/Dialect/Bufferization/Transforms/one-shot-bufferize.mlir

```diff
@@ -268,4 +268,10 @@ func.func @materialize_in_dest_raw(%f: f32, %f2: f32, %idx: index) -> (tensor<5x
   %r = tensor.extract %dest_filled[%idx] : tensor<5xf32>
 
   return %0, %r : tensor<5xf32>, f32
-}
\ No newline at end of file
+}
+
+// -----
+// CHECK-LABEL: @llvm_return
+func.func @llvm_return() {
+   llvm.return
```

Comment:
Do you actually have `llvm.return` ops in your use case or do you have a custom return op? This example looks odd to me because it mixes `func.func` with `llvm.return`. I'm surprised that this is even valid IR.

Reply:
This case is from https://github.com/llvm/llvm-project/issues/120535.

A `func.func` with `llvm.return` is legal IMO. `func.return` doesn't check parent op in verifier and this is common in a conversion with mannul selected patterns.

Reply:
Bufferization should be done before lowering to LLVM. Tensor types are not valid LLVM types.

Reply:
If this is just meant to avoid the crash, I would replace this like:
```c++
  assert(!returnOps.empty() && "expected at least one ReturnOp");
```

With `return failure();` and maybe `emitOpError`.

Also note the name of one of the files: `FuncBufferizableOpInterfaceImpl.cpp`. This is a func-dialect specific implementation.